Dockerfile

```diff
@@ -1,11 +1,82 @@
-ARG DEBIAN_DIST=bullseye
+# syntax=docker/dockerfile:1.10
```

Comment:
This is a complete rewrite of the dockerfile

Can any of this be broken out into smaller (complete) steps (ie commits) to make this easier to follow along?